cmake/CMakeLists.txt

```
@@ -642,6 +643,22 @@ if(BUILD_TOOLS)
   install(FILES ${LAMMPS_DOC_DIR}/msi2lmp.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
 endif()
 
+if(BUILD_LAMMPS_SHELL)
+  include(FindPkgConfig)
+  pkg_check_modules(READLINE IMPORTED_TARGET readline)
+  if(PKG_CONFIG_FOUND AND READLINE_FOUND)
+    if(NOT LAMMPS_EXCEPTIONS)
+      message(WARNING "The LAMMPS shell needs LAMMPS_EXCEPTIONS enabled for full functionality")
+    endif()
+    add_executable(lammps-shell ${LAMMPS_TOOLS_DIR}/lammps-shell/lammps-shell.cpp)
+    target_compile_definitions(lammps-shell PRIVATE -DLAMMPS_LIB_NO_MPI)
+    target_link_libraries(lammps-shell PRIVATE lammps;PkgConfig::READLINE)
+    install(TARGETS lammps-shell DESTINATION ${CMAKE_INSTALL_BINDIR})
```

```suggestion
    install(TARGETS lammps-shell EXPORT LAMMPS_Targets DESTINATION ${CMAKE_INSTALL_BINDIR})
```
export the target as well.